### PR TITLE
Ifrost/global header tweaks

### PR DESCRIPTION
**Pull request by ifrost1**
### Summary:
- Adjusted `UtilityNav` to be more composable. Popover shouldn't be directly built into the `UtilityNav` for future flexibilty for adding more items (etc. notifications and an account link).

Addressed the conversation we had with Courtney and Sean:
- Hover states for primary nav
- Break into a top nav bar on larger screens so List Detail doesn't squish
- Visually hide the name from the Avatar in `UtilityNavItem`. I added the name `Notifications` for that since it technically triggers notifications and a name doesn't serve a purpose there.

### Test Plan:
 Ran `yarn test`. It ran with a deprecated `variant="dark` message that doesn't have anything to do with my work

`yarn lint` passed

`yarn start` spins up.

### CODE REVIEW

File: .storybook/recipes/GlobalHeader/GlobalHeader.module.css [hunk 1]
```diff
@@ -42,27 +42,41 @@
...
  */
-.global-header__avatar-block {
+.global-header__utility-nav {
   position: relative;
-  left: -0.5rem; /* 2 */
   margin-left: auto;
 
-  @media all and (min-width: $eds-bp-lg) {
+  @media all and (min-width: $eds-bp-xl) {
+    left: -0.5rem;
```
Comment (anniehu4) [inline .storybook/recipes/GlobalHeader/GlobalHeader.module.css:66]: the logo & utility nav look a little too far left of the PrimaryNav:
Screen Shot 2022-04-07 at 10 59 08 AM

Is this intentional? Or should we remove the `left:` styling from them?

File: src/components/PrimaryNav/PrimaryNav.module.css [hunk 2]
```diff
@@ -36,6 +41,18 @@
   @mixin TextLinkInverted;
   text-decoration: none;
   white-space: nowrap;
+  padding: 0.5rem;
```
Comment (anniehu4) [inline src/components/PrimaryNav/PrimaryNav.module.css:44]: nit
```suggestion
  padding: var(--eds-size-1);
```

File: .storybook/recipes/GlobalHeader/GlobalHeader.module.css [hunk 3]
```diff
@@ -42,27 +42,41 @@
...
  *    can be exposed
  */
-.global-header__menu-button {
-  @media all and (min-width: $eds-bp-lg) {
+.global-header__menu-button.global-header__menu-button {
+  transition: background var(--eds-anim-fade-quick) var(--eds-anim-ease);
+
+  &:hover,
+  &:focus {
+    background: var(--eds-color-brand-grape-800);
```
Comment (anniehu4) [inline .storybook/recipes/GlobalHeader/GlobalHeader.module.css:50]:
```suggestion
    background: var(--eds-theme-color-text-link-strong-hover);
```

File: src/components/PrimaryNav/PrimaryNav.module.css [hunk 4]
```diff
@@ -36,6 +41,18 @@
...
+  border-radius: var(--eds-theme-border-radius);
+  transition: background var(--eds-anim-fade-quick) var(--eds-anim-ease);
+
+  &:hover,
+  &:focus {
+    background: var(--eds-color-brand-grape-800);
```
Comment (anniehu4) [inline src/components/PrimaryNav/PrimaryNav.module.css:50]:
```suggestion
    background: var(--eds-theme-color-text-link-strong-hover);
```

File: src/components/UtilityNav/UtilityNav.tsx [hunk 5]
```diff
@@ -29,13 +29,13 @@ export interface Props {
 /**
  * Primary UI component for user interaction
  */
-export const UtilityNav = ({
+export const UtilityNav: React.FC<Props> = ({
```
Comment (anniehu4) [inline src/components/UtilityNav/UtilityNav.tsx:32]: nit: can revert this change

File: src/components/UtilityNavItem/UtilityNavItem.tsx [hunk 6]
```diff
@@ -67,28 +88,32 @@ export const UtilityNavItem = React.forwardRef<HTMLLIElement, Props>(
...
           )}
+          <span
+            className={
+              hideText
+                ? `${styles['utility-nav__text']} ${utilityStyles['u-is-vishidden']}`
+                : styles['utility-nav__text']
+            }
+          >
+            {text}
+          </span>
```
Comment (anniehu4) [inline src/components/UtilityNavItem/UtilityNavItem.tsx:116]: Currently we prefer using an `aria-label` over `hideText`. We're looking into the i18n a11y concerns Brad brought up as well, but this at least keeps it consistent

File: .storybook/recipes/GlobalHeader/GlobalHeader.module.css (hunk 1 above)
Comment (ifrost1) [reply .storybook/recipes/GlobalHeader/GlobalHeader.module.css:66]: I added some padding for the hover states to the primary nav items. I can tighten this alignment stuff up

File: .storybook/recipes/GlobalHeader/GlobalHeader.module.css (hunk 3 above)
Comment (ifrost1) [reply .storybook/recipes/GlobalHeader/GlobalHeader.module.css:50]: Courtney and I were talking about how to handle these. `text-link` refers to actual text. Because this is technically a background, we may want to use the `button-icon` token for this. One of those weird background hover examples.

File: src/components/PrimaryNav/PrimaryNav.module.css (hunk 4 above)
Comment (ifrost1) [reply src/components/PrimaryNav/PrimaryNav.module.css:50]: Courtney and I were talking about how to handle these. text-link refers to actual text. Because this is technically a background, we may want to use the button-icon token for this. One of those weird background hover examples.

File: .storybook/recipes/GlobalHeader/GlobalHeader.tsx [hunk 7]
```diff
@@ -34,6 +43,22 @@ export const GlobalHeader = ({ className, ...other }: Props) => {
...
+  useEffect(() => {
+    updateScreenSize();
+    window.addEventListener('resize', updateScreenSize);
+    return () => {
+      window.addEventListener('resize', updateScreenSize, false);
+    };
+  }, []);
+
+  const updateScreenSize = () => {
+    if (window.innerWidth >= 1200) {
```
[the remark this answers is not in the document]
Comment (ifrost1) [reply .storybook/recipes/GlobalHeader/GlobalHeader.tsx:55]: Happy to try that out. That's a good callout

File: .storybook/recipes/GlobalHeader/GlobalHeader.tsx [hunk 8]
```diff
@@ -34,6 +43,22 @@ export const GlobalHeader = ({ className, ...other }: Props) => {
     }
   };
 
+  useEffect(() => {
+    updateScreenSize();
+    window.addEventListener('resize', updateScreenSize);
+    return () => {
+      window.addEventListener('resize', updateScreenSize, false);
```
[the remark this answers is not in the document]
Comment (ifrost1) [reply .storybook/recipes/GlobalHeader/GlobalHeader.tsx:50]: Thanks for this. That's what I get for copy and pasting.

File: src/components/UtilityNavItem/UtilityNavItem.tsx (hunk 6 above)
Comment (ifrost1) [reply src/components/UtilityNavItem/UtilityNavItem.tsx:116]: Happy to do whatever makes you all feel good as long as it's consistent. Utility classes are something I'm curious about in terms of CSS modules. Seems like we'd want those to be global styles you can pass in throughout the library and people can use them without having to import them. I'll admit I'm a little less experienced with CSS modules so still learning the nuance

File: .storybook/recipes/GlobalHeader/GlobalHeader.module.css (hunk 3 above)
Comment (ifrost1) [reply .storybook/recipes/GlobalHeader/GlobalHeader.module.css]: Added a TODO for these until we come to a clear decision with design around this

File: src/components/UtilityNavItem/UtilityNavItem.tsx (hunk 6 above)
Comment (ifrost1) [reply src/components/UtilityNavItem/UtilityNavItem.tsx:116]: @anniehu4 Do you want this changed to aria label or should we leave it for now until we find a more permanent solution? Happy to do either. Thanks!

**Review by anniehu4 — APPROVED** [review]

File: src/components/UtilityNavItem/UtilityNavItem.tsx (hunk 6 above)
Comment (anniehu4) [reply src/components/UtilityNavItem/UtilityNavItem.tsx]: > Utility classes are something I'm curious about in terms of CSS modules. Seems like we'd want those to be global styles you can pass in throughout the library and people can use them without having to import them. I'll admit I'm a little less experienced with CSS modules so still learning the nuance

Yeah, that makes sense to me. I'm thinking we could export those as regular `.css` files and then import at a top-level (e.g. `preview.js` for storybook, `index.js` for the App) so they're available globally
